### ISSUE TITLE
Uploaded file objects are not seeked to 0

### ISSUE THREAD

**Issue by jonashaag** [gradio-app/gradio#2878]
### Describe the bug

I have this piece of code to copy and uploaded file somewhere else:

```
def example(uploaded_file):
    ...
    uploaded_file.seek(0)
    shutil.copyfileobj(uploaded_file, target_file)
    ...

demo = gr.Interface(example, "file", "text")
```

Without the call to `.seek(0)` `copyfileobj` doesn't copy anything, so I assume `uploaded_file` is not 0-seeked.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

See above

### Logs

```shell
-
```

### System Info

```shell
gradio               3.15.0
```

### Severity

annoying

**Comment by abidlabs** [comment]
Very interesting, thanks @jonashaag. Would you be open to making a PR for this? Looks like the fix should be in the `File` component (https://github.com/gradio-app/gradio/blob/1765cf68ae7181093db3e0e5c4df33797557e22b/gradio/components.py#L2145), and perhaps some of the other file-based components as well.

**Comment by abidlabs** [comment]
I tested this and this is no longer present as an issue in `gradio==3.20.0`. Presumably, it was fixed as a side effect of #3191

Test code:

```py
import shutil
import gradio as gr

def example(uploaded_file):
    with open(uploaded_file.name, "r") as input:
        with open("test_location.txt", "w") as output:
            shutil.copyfileobj(input, output)
    return "Done"

demo = gr.Interface(example, "file", "text")
f = gr.File() 
t = gr.Text()
    
gr.Interface(example, f, t).launch()
```

Upload a `.txt` file and look at `test_location.txt` -- you'll see that the file has been copied over fully.
